### ISSUE TITLE
Add Search filter (by ean/upc/location) for Stock update

### ISSUE THREAD

**Issue by Ybeske** [PrestaShop/PrestaShop#10875]
I really miss the feature to search by ean code, upc code and product location in the search bar for stock management - stock update movements.

"search by ean-/upc code": When the products arrive, the need to search by ean and upc code - using a barcode scanner- will be very user friendly to easily find the right product to update!

"search by product location": the search filter by product location can simplify the management of the stock locations. For example: When I'm wondering rather a physically empty location is assigned to an active product, the search by location will really be helpful! Also for inventory matters.

![search filter stock update](https://user-images.githubusercontent.com/43880279/46536328-748c0c00-c8ae-11e8-8678-e11c47259fec.jpg)

**Comment by khouloudbelguith** [comment]
Hi @Ybeske,

Thanks for your improvement.
Related to: #9727.
@marionf what do you think?
Thanks!

**Comment by Ybeske** [comment]
Hi,

@khouloudbelguith 
@marionf

I really hope the utility of my proposal triggerd some thoughts :-)
Can you please inform me about a possible implementation of this feature in the coming version 1.7.5?

Thanks!

**Comment by khouloudbelguith** [comment]
@Ybeske, the new version 1.7.5.0 will be released in a few weeks, we only accept security fixes or regression from 1.7.4.0 on this branch to be focused on the 1.7.5.0 version, as described here: https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#target-branch
Thanks!

**Comment by MatShir** [comment]
Hi @Ybeske,

For now, the product references (also called SKU) is used to help the merchant to find the product in the Front Office or the Back Office. This homemade reference is composed of the abbreviation of the localization, type of product, size, so and so forth. 
For example:
Summer 2012, Gecko Tee, Small, Red, Shop 1, Top shelf = S12-Gec-RS-1T
It might not be the most trivial product searching. We got your feedback and are considering it. We also have shared the idea at Prestashop Feedbacks (http://feedback.prestashop.com) to let help us to prioritize all ideas from the community.

Thank you

**Comment by Ybeske** [comment]
Thanks for the idea to create this alternative way of product search by location. 
I can read on the screenshot the search engine only search the 3 shown fields (name, ref and supplier). The ref (or sku) is also the unique code to communicate with the customers. I really would like to keep these string cleaner and recognizable and use the propriate fields to search the products.
However I suggest the search filter on this page (see screenshot) at the stock movements (stock manager) to simply search also in the existing location field, the existing ean field and the existing upc field of each product. 
In your alternative way you create an extra/double management of all these info fields if you want the search to be posible by the ref field...
Please inform me if I am wrong. 😉

**Comment by MatShir** [comment]
Yes, of course. The reference product is here only to find the product easily (short and practical Alphanumeric) without knowing the location or UPC but it will not be optimized for specific research. I was a small tip for urgent needs. It will be discussed with the team.
Thank you for your feedbacks

**Comment by omar2886** [comment]
This solution would be very useful actually. In the meantime, the searchbox at the top of the page can also search for EAN and give you the product as result, the only problem is that users with "salesman" permission won't have access to it, wouldn't it be faisable to give them access to this searbox? they won't get any results for things they don't have permission anyway, am I wrong?

**Comment by Ybeske** [comment]
Yes, I know you can search on the top for products and edit the product itself, you can change (manually calculated) the new stock amount. If you open this product page in the back office, you can overwrite the new amount. If you wait some minutes (distracted by a call, ...) you can still save. This will overwrite the previously loaded amount on that moment. BUT: If - by coincidence - somebody bought that product during your call, you will have stock errors. The wrong amount will be overwritten in the database without taking into account the bought amount. The amount you had is a situation on the moment of page loading in the back office.
It seems you can't have stock errors by updating the product quantity with the stock manager (movements). Here you update the amount increasing of decreasing the actual amount.
Having more experience I also/rather would like to have the search by "supplier code" in the stock manager menu.
Why are it these suggested fields (ean, upc, location, suppier code, ...) not searchable like they can be found in the upper search box?
Hope to here from you.

Thanks in advance!

**Comment by Ybeske** [comment]
Can someone please inform me about the probability of this feature (or improvement) and from what PS version?

**Comment by pollat** [comment]
So, has EAN, UPC or MPN research on inventory management been implemented?

**Comment by florine2623** [comment]
For EAN and UPC it works on PS 9.0.0.

Search by product location is not implemented yet.

**Comment by boo-code** [comment]
Opened as #42454, as a draft.

Worth saying first that two thirds of this already work. `QueryParamsCollection::appendSqlSearchFilter()` matches the search box against product and combination EAN, ISBN, **UPC** and MPN as well as the reference, both names and the supplier, so scanning an EAN or a UPC on Sell > Stock finds the row on 9.2.x today. If that is not what you see, it is a different defect and worth its own report.

The location genuinely was missing. It is absent from the search field list, has no entry in the token to column map, and neither stock query selected it, although both already join `stock_available`. The PR selects `COALESCE(sa.location, "")` as `product_location` in both, maps the token and adds the field. `stock_available.location` rather than `product.location`, so a combination keeps its own value.

One limitation stated plainly: the search fields are shared with the movements list, so the column is selected by both queries. I could not test that half. A request to `api_stock_list_movements` with a keyword still returns 200 with the movements query deliberately broken, so it never reaches it under the current integration harness, and I removed the test rather than keep one that proves nothing.
